src/main/java/de/tum/in/www1/artemis/service/LectureUnitImportService.java

```diff
@@ -0,0 +1,173 @@
+package de.tum.in.www1.artemis.service;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Attachment;
+import de.tum.in.www1.artemis.domain.Lecture;
+import de.tum.in.www1.artemis.domain.lecture.AttachmentUnit;
+import de.tum.in.www1.artemis.domain.lecture.ExerciseUnit;
+import de.tum.in.www1.artemis.domain.lecture.LectureUnit;
+import de.tum.in.www1.artemis.domain.lecture.OnlineUnit;
+import de.tum.in.www1.artemis.domain.lecture.TextUnit;
+import de.tum.in.www1.artemis.domain.lecture.VideoUnit;
+import de.tum.in.www1.artemis.repository.AttachmentRepository;
+import de.tum.in.www1.artemis.repository.LectureUnitRepository;
+import de.tum.in.www1.artemis.repository.iris.IrisSettingsRepository;
+import de.tum.in.www1.artemis.service.connectors.pyris.PyrisWebhookService;
+
+@Profile(PROFILE_CORE)
+@Service
+public class LectureUnitImportService {
+
+    private static final Logger log = LoggerFactory.getLogger(LectureUnitImportService.class);
+
+    private final LectureUnitRepository lectureUnitRepository;
+
+    private final AttachmentRepository attachmentRepository;
+
+    private final FileService fileService;
+
+    private final SlideSplitterService slideSplitterService;
+
+    private final Optional<PyrisWebhookService> pyrisWebhookService;
+
+    private final Optional<IrisSettingsRepository> irisSettingsRepository;
+
+    public LectureUnitImportService(LectureUnitRepository lectureUnitRepository, AttachmentRepository attachmentRepository, FileService fileService,
+            SlideSplitterService slideSplitterService, Optional<PyrisWebhookService> pyrisWebhookService, Optional<IrisSettingsRepository> irisSettingsRepository) {
+        this.lectureUnitRepository = lectureUnitRepository;
+        this.attachmentRepository = attachmentRepository;
+        this.fileService = fileService;
+        this.slideSplitterService = slideSplitterService;
+        this.pyrisWebhookService = pyrisWebhookService;
+        this.irisSettingsRepository = irisSettingsRepository;
+    }
+
+    /**
+     * This function imports the lecture units from the {@code importedLecture} and appends them to the {@code lecture}
```

Comment:
Is `@code` the right tag here? Maybe `@link`?

Reply:
They are no classes or methods we can link to. They only refer to the parameter names.